tests/cli/test_suite.py

```diff
@@ -1,489 +1,1109 @@
+
+  Expectation Suite operations
+
+Options:
+  --help  Show this message and exit.
+
+Commands:
+  delete  Delete an expectation suite from the expectation store.
+  demo    This command is not supported in the v3 (Batch Request) API.
+  edit    Generate a Jupyter notebook for editing an existing Expectation...
+  list    Lists available Expectation Suites.
+  new     Create a new empty Expectation Suite.
+"""
+        in stdout
     )
     assert_no_logging_messages_or_tracebacks(
         my_caplog=caplog,
         click_result=result,
     )
 
 
-def test_suite_demo_deprecation_message(caplog, monkeypatch, empty_data_context):
-    runner = CliRunner(mix_stderr=False)
-    monkeypatch.chdir(os.path.dirname(empty_data_context.root_directory))
-    result = runner.invoke(cli, ["--v3-api", "suite", "demo"], catch_exceptions=False)
-    assert result.exit_code == 0
-    assert (
-        "This command is not supported in the v3 (Batch Request) API." in result.stdout
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
+)
+def test_suite_demo_deprecation_message(
+    mock_emit, caplog, monkeypatch, empty_data_context_stats_enabled
+):
+    context: DataContext = empty_data_context_stats_enabled
+
+    monkeypatch.chdir(os.path.dirname(context.root_directory))
+
+    runner: CliRunner = CliRunner(mix_stderr=False)
+    result: Result = runner.invoke(
+        cli,
+        f"--v3-api suite demo",
+        catch_exceptions=False,
     )
+    assert result.exit_code == 0
+
+    stdout: str = result.stdout
+    assert "This command is not supported in the v3 (Batch Request) API." in stdout
+
+    expected_call_args_list = [
+        mock.call(
+            {"event_payload": {}, "event": "data_context.__init__", "success": True}
+        ),
+        mock.call(
+            {
+                "event": "cli.suite.demo.begin",
+                "event_payload": {"api_version": "v3"},
+                "success": True,
+            }
+        ),
+        mock.call(
+            {
+                "event": "cli.suite.demo.end",
+                "event_payload": {"api_version": "v3"},
+                "success": True,
+            }
+        ),
+    ]
+
+    assert mock_emit.call_args_list == expected_call_args_list
+    assert mock_emit.call_count == len(expected_call_args_list)
+
     assert_no_logging_messages_or_tracebacks(
         my_caplog=caplog,
         click_result=result,
     )
 
 
-@pytest.mark.xfail(
-    reason="This command is not yet implemented for the modern API",
-    run=True,
-    strict=True,
+@mock.patch(
+    "great_expectations.core.usage_statistics.usage_statistics.UsageStatisticsHandler.emit"
 )
 @mock.patch("subprocess.call", return_value=True, side_effect=None)
 @mock.patch("webbrowser.open", return_value=True, side_effect=None)
-def test_suite_new_creates_empty_suite(
+def test_suite_new_non_interactive_with_suite_name_prompted_default_opens_jupyter(
```

Comment:
How about changing the name of this test since it runs the notebook and generates a suite?

Reply:
Great idea -- done in the next commit.